### PR TITLE
fix ingest warnings

### LINKED ISSUE #2554: No error message displayed on json load exception (regression after GA Brim v0.28.0)
Before opening a new issue, please make sure you've reviewed the troubleshooting guide:
https://github.com/brimdata/brim/wiki/Troubleshooting

**Describe the bug**
No warning message displayed on json load exception
Version 0.26 normally indicates a parsing error
**To Reproduce**
Loads a multi-line json file that has lines that cannot be parsed properly.
Brim only parses the data before the exception line, and there is no error message.

**Expected behavior**
Display a warning message about an engine parsing error.

**Screenshots**

**Desktop Info**
 - OS: Windows
 - Brim Version: 0.31.0

### LINKED ISSUE #2606: No error message displayed on CSV load exception
Repro is with Brim commit 96a0951 which uses Zed commit `22f640e`. This issue feels very similar to the existing JSON-related one in #2554.

A community zync user reported a problem reading CSV with Zui. In their own words:

> Zui did not report any problem. It just stopped the CSV import without any message at all. I noticed because I compared `count()` to the `wc`.
> maybe simply send a more comprehensive error message in Zed/Zui when CSV import fails before the EOF? line number and column number with last value parsed?

Indeed, as shown in the following video, reading the file [bad.csv](https://github.com/brimdata/brim/files/10158482/bad.csv) using CLI tooling does show an error, but the app just shows the data that parsed ok without any mention of the error. This is a data correctness hazard since it gives the user the impression they're working with all their data when it's actually truncated.

https://user-images.githubusercontent.com/5934157/205760538-de1072b7-47da-4c75-82c3-0a9f72d38488.mp4

As we can see in Wireshark, the error message was returned from the backend, but just didn't make it all the way to the user.

image

### PR DESCRIPTION
**Pull request by nwt**
Ingest warnings have been broken for a while.  Fix them.

Included are fixes to some global dispatcher bugs that were causing the Pools.appendWarning reducer to run three times per dispatch.

The video below shows what ingest warnings look like now.

https://user-images.githubusercontent.com/2574448/217313236-44ebf753-2e5c-4545-abba-5af486c24bf0.mov

Closes #2554
Closes #2606

### CODE REVIEW

File: src/js/components/IngestWarningsModal.tsx [hunk 1]
```diff
@@ -31,7 +31,11 @@ export default function IngestWarningsModal({onClose}) {
       <Title>Ingest Warnings</Title>
       {warnings.length ? (
         <>
-          <p>{"Data ingest failed with these warnings."}</p>
+          <p>
+            {
+              "The data you've attempt to import was not recognized as any supported packet capture or log format. The errors returned by each log parser:"
```
Comment (jameskerr) [inline src/js/components/IngestWarningsModal.tsx:36]:
```suggestion
              "The data you've attempted to import was not recognized as any supported packet capture or log format. The errors returned by each log parser:"
```

**Review by jameskerr — APPROVED** [review]
One spelling error, then good to merge.

**Comment by philrz** [comment]
It's definitely a huge improvement that the warnings are now visible to the user, so I'd absolutely take this fix as is. That said, given that "load" and "ingest" are roughly synonymous, it feels strange that the messages "Ingest failed with warnings" and "Load successful" appear on the screen at the same time. If it were possible in this case to change the "Load successful" to something that reflects the load only partially succeeded, that would help a lot, I think.

**Comment by nwt** [comment]
@philrz: I agree, but that "Load successful" toast comes from code I'm not touching here, and the way to make the change you suggest isn't obvious to me, so I think that's fodder for another PR.

**Comment by philrz** [comment]
Thanks @nwt. I've opened #2660 so we can revisit that in the future.

**Pull request merged**